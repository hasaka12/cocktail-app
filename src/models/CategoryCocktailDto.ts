export interface CategoryDto {
  strDrink: string;
  strDrinkThumb: string;
  idDrink: string;
}

export interface CategoryCocktailDto {
  drinks: CategoryDto[];
  key: string;
}
