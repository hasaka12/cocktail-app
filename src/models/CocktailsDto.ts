export interface CocktailsDto {
  idDrink: string;
  strAlcoholic: string;
  strCategory: string;
  strDrink: string;
  strDrinkThumb: string;
  strGlass: string;
  strInstructions: string;
}
