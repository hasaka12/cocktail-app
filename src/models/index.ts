export * from "./CocktailsDto";
export * from "./SearchCocktailDto";
