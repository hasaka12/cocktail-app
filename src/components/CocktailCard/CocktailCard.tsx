import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import Typography from "@mui/material/Typography";
import StarIcon from "@mui/icons-material/Star";
import Button from "@mui/material/Button";

import { CocktailsDto } from "../../models";

export interface CocktailCardProps {
  title: string;
  desc?: string;
  isAddedToFavorite?: boolean;
  cocktail?: CocktailsDto;
  handleAddFavorite?: (cocktail: CocktailsDto) => void;
  img: string;
}

const CocktailCard = ({
  title,
  desc,
  img,
  isAddedToFavorite,
  cocktail,
  handleAddFavorite,
}: CocktailCardProps) => {
  return (
    <Card sx={{ maxWidth: 345 }}>
      <CardMedia component="img" height="140" image={img} alt={desc} />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {title}
        </Typography>
        {desc ? (
          <Typography
            variant="body2"
            color="text.secondary"
            id="description-text"
          >
            {desc}
          </Typography>
        ) : null}
      </CardContent>
      {cocktail ? (
        <div className="m-2">
          <Button
            variant="contained"
            fullWidth
            onClick={() => {
              handleAddFavorite && cocktail && handleAddFavorite(cocktail);
            }}
            endIcon={
              <StarIcon color={isAddedToFavorite ? "warning" : "inherit"} />
            }
          >
            {isAddedToFavorite ? "Remove from Favorites" : "Add to Favorites"}
          </Button>
        </div>
      ) : null}
    </Card>
  );
};

export default CocktailCard;
