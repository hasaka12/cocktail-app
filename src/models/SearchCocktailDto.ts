import { CocktailsDto } from "./CocktailsDto";

export interface SearchCocktailDto {
  drinks: CocktailsDto[];
}
