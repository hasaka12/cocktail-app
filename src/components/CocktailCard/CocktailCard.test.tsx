import { render, screen, act } from "@testing-library/react";
import user from "@testing-library/user-event";

import CocktailCard from "./CocktailCard";
import { CocktailCardProps } from "./CocktailCard";
import { CocktailsDto } from "../../models";

const cocktail: CocktailsDto = {
  idDrink: "123",
  strAlcoholic: "Mojito",
  strCategory: "Alcoholic",
  strDrink: "Drink",
  strDrinkThumb: "thumb",
  strGlass: "Glass",
  strInstructions: "Instructions",
};

describe("CocktailCard", () => {
  test("renders CocktailCard component with title and description", () => {
    const props: CocktailCardProps = {
      title: "Mojito",
      desc: "A refreshing cocktail",
      img: "mojito.jpg",
      isAddedToFavorite: false,
      cocktail,
      handleAddFavorite: jest.fn(),
    };

    render(<CocktailCard {...props} />);

    expect(screen.getByText("Mojito")).toBeInTheDocument();
    expect(screen.getByText("A refreshing cocktail")).toBeInTheDocument();
  });

  test("renders CocktailCard component without description", () => {
    const props: CocktailCardProps = {
      title: "Martini",
      desc: "",
      img: "martini.jpg",
      isAddedToFavorite: false,
      cocktail,
      handleAddFavorite: jest.fn(),
    };

    render(<CocktailCard {...props} />);

    expect(screen.getByText("Martini")).toBeInTheDocument();
    expect(screen.queryByTestId("description-text")).toBeNull();
  });

  test('clicking on "Add to Favorites" button calls handleAddFavorite function with the cocktail', () => {
    const props: CocktailCardProps = {
      title: "Margarita",
      desc: "A classic cocktail",
      img: "margarita.jpg",
      isAddedToFavorite: false,
      cocktail,
      handleAddFavorite: jest.fn(),
    };

    render(<CocktailCard {...props} />);
    const addFavButton = screen.getByRole("button", {
      name: /add to Favorites/i,
    });

    act(() => {
      user.click(addFavButton);
    });

    expect(props.handleAddFavorite).toHaveBeenCalledWith(props.cocktail);
  });

  test("Renders correctly", () => {
    const props: CocktailCardProps = {
      title: "Mojito",
      desc: "A refreshing cocktail",
      img: "mojito.jpg",
      isAddedToFavorite: false,
      cocktail,
      handleAddFavorite: jest.fn(),
    };

    const { asFragment } = render(<CocktailCard {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
