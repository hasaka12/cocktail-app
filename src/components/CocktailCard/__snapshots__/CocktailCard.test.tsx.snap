// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailCard Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="A refreshing cocktail"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="140"
      src="mojito.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Mojito
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        id="description-text"
      >
        A refreshing cocktail
      </p>
    </div>
    <div
      class="m-2"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Add to Favorites
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="StarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
